eng/scripts/automation/GenerateAndBuildLib.ps1

```diff
@@ -883,4 +904,31 @@ function UpdateExistingSDKByInputFiles()
         GeneratePackage -projectFolder $projectFolder -sdkRootPath $sdkRootPath -path $path -downloadUrlPrefix "$downloadUrlPrefix" -serviceType $serviceType -generatedSDKPackages $generatedSDKPackages
     }
 
+}
+
+function GetSDKProjectFolder()
+{
+    param(
+        [string]$typespecConfigurationFile,
+        [string]$sdkRepoRoot
+    )
+    $tspConfigYaml = Get-Content -Path $typespecConfigurationFile -Raw
+
+    Install-ModuleIfNotInstalled "powershell-yaml" "0.4.1" | Import-Module
+    $yml = ConvertFrom-YAML $tspConfigYaml
+    $service = ""
+    $packageDir = ""
+    if ($yml) {
+        if ($yml["parameters"] -And $yml["parameters"]["service-dir"]) {
+            $service = $yml["parameters"]["service-dir"]["default"];
+        }
+        if ($yml["options"] -And $yml["options"]["@azure-tools/typespec-csharp"] -And $yml["options"]["@azure-tools/typespec-csharp"]["package-dir"]) {
+            $packageDir = $yml["options"]["@azure-tools/typespec-csharp"]["package-dir"]
+        }
+    }
+    if (!$service || !$packageDir) {
+        throw "Not provide service name or packageDir."
```

Comment:
Can you include instructions in the message to guide users what they should do to resolve this issue?

Reply:
We may file another issue for all error-how-to-fix information.